### DISCUSSION TITLE
registering toggle_quick_terminal with KGlobalAccel for kde plasma shortcut?

### DISCUSSION THREAD

**Discussion by digitalsignalperson** [ghostty-org/ghostty#3192] [Feature Requests, Ideas] [answered]
On kde plasma with wayland, it would be cool to have the quake style global shortcut. 

Example with yakuake registering shortcut with KGlobalAccel:
https://github.com/KDE/yakuake/blob/164d24b8bad1175199260c627be30d623802465c/app/mainwindow.cpp#L254

Dunno if there's a way to achieve this. Global shortcuts with wayland seem in an awkward state here.

**Comment by Skatox** [comment]
Agree, this is the single feature stopping me from using ghostty

**Comment by rhodes-b** [comment]
the quick terminal is a macos only feature

on linux it is a GTK based app which does not support this feature. There may be a QT based implimentation in the future with `libghostty`
which would be maintained by the community
https://ghostty.org/docs/about#libghostty

but GTK will be the only officially supported linux version

**Reply by digitalsignalperson** [reply]
It might still be possible with GTK somehow just interfacing over dbus via org.kde.KGlobalAccel. Poking around required.

Alternately, someone could work on a kwin script to show/hide at the top of the screen the mouse cursor is currently on (I think that is all feasible, have done similar before).

**Reply by jcollie** [reply]
The Ghostty GTK apprt strives to be a native GTK app, so interfacing with KDE DBus APIs is unlikely to happen (IMHO). KDE features are best implemented in a native Qt apprt.

**Reply by rhodes-b** [reply]
> It might still be possible with GTK somehow just interfacing over dbus via org.kde.KGlobalAccel. Poking around required.
> 
> Alternately, someone could work on a kwin script to show/hide at the top of the screen the mouse cursor is currently on (I think that is all feasible, have done similar before).

You are free to fork ghostty in the meantime if you want to add this feature yourself (jcollie explained why we wouldnt support a KDE specific feature)

**Comment by simontong** [comment]
This is sadly a deal breaker for me (being a KDE user). This one feature would make me switch from Tilix for sure.

**Comment by jcollie** [comment, marked as answer]
There's hope yet: #4624

**Reply by mitchellh** [reply]
I recommend adding any notes to this issue.

**Reply by digitalsignalperson** [reply]
I think the original question still remains, how to actually use this on linux with kde

**Reply by pluiedev** [reply]
Quick terminal itself already exists but registering global shortcuts is still not there yet. There's an open issue about it

**Reply by digitalsignalperson** [reply]
@pluiedev ah thanks found it: #6051. I'll subscribe there
